### PR TITLE
Added checks for transient properties

### PR DESCRIPTION
**Pull request by lrrrs**
As I mentioned, I'm using transient properties quite a lot in my app, which are causing lots of empty events, when the context is saved. This change discards empty changes.

Some unit tests are not working right now, because this change is reducing the number of events in some cases. I haven't had the time to check each test yet, but maybe you can give me a hint if I'm on the right track here ;)

### CODE REVIEW

**Comment by drewmccormack** [comment]
It looks like it could work. It is important to bail before the save event is actually created and saved. I'm not sure if you have been able to do that. Would need to look closely at the CDEEvemtBuilder code.

Once the save event is saved, it influences any merge that may happen at the same time. Removing it after it is saved could cause a missing revision number.

As long as you can be sure you bail before the save event is saved, it should work.

File: Framework/Source/Events/CDESaveMonitor.m [hunk 1]
```diff
@@ -197,6 +202,12 @@ - (void)asynchronouslyStoreChangesForContext:(NSManagedObjectContext *)context c
     NSDictionary *deleteData = [eventBuilder changesDataForDeletedObjects:deletedObjects inManagedObjectContext:context];
     [changedValuesByContext removeObjectForKey:context];
 
+    if(((NSArray*)insertData[@"changeArrays"]).count + ((NSArray*)updateData[@"objectIDs"]).count + ((NSArray*)deleteData[@"orderedObjectIDs"]).count == 0)
```
Comment (lrrrs) [inline Framework/Source/Events/CDESaveMonitor.m]: Bail here, if no changes have been made and deregister the IncompleteEventIdentifier

Comment (drewmccormack) [inline Framework/Source/Events/CDESaveMonitor.m]: Yeah, it looks like the event is created just under that, so bailing there should be fine.

**Comment by lrrrs** [comment]
All tests green now ;)

**Pull request merged**